### DISCUSSION TITLE
Improve/add PostgreSQL TYPE support

### DISCUSSION THREAD

**Discussion by jenstroeger** [sqlalchemy/alembic#1157] [Ideas]
Looking at Alembic’s [Operation Reference](https://alembic.sqlalchemy.org/en/latest/ops.html) I noticed that there’s no support for [`CREATE TYPE`](https://www.postgresql.org/docs/15/sql-createtype.html) or [`ALTER TYPRE`](https://www.postgresql.org/docs/15/sql-altertype.html) or [`DROP TYPE`](https://www.postgresql.org/docs/15/sql-droptype.html).

Why’s that? Are there plans to add support for those?

**Comment by CaselIT** [comment]
Hi,

Those are postgresql only, not very used, have many different forms (so the api would be a bit messy, since you would for example have many kwargs that are mutually exclusive), and have strange limitations (like add enum value for the longest time required autocommit, and even in current pg versions has limitations) that complicate the use with autogenerate.

Overalls for these cases using the text with `execute` is probably clearer

**Comment by jenstroeger** [comment]
@CaselIT I suspected as much and that makes sense. Thank you for the clarification and confirmation.

**Reply by CaselIT** [reply]
There an external package thet has some additional functionality https://github.com/olirice/alembic_utils

Not sure if it covers this case though

**Comment by zzzeek** [comment]
We would like to support TYPE operations, at least regarding ENUM, once we finally (if ever) address the overall [enum issue](https://github.com/sqlalchemy/alembic/labels/autogenerate%20for%20enums) in a holistic fashion.   until then I've just not been moving on anything to do with TYPE so that I dont paint myself into a corner with only partial solutions.
